### PR TITLE
jobs: refactor combo tracker as an event emitter

### PR DESCRIPTION
**Pull request by MaikoTan**
Also fix a bug that combo timer would just stop after 15 seconds, because there is a `setTimeout` called with the old 15000ms.

### CODE REVIEW

File: ui/jobs/combo_tracker.ts [hunk 1]
```diff
@@ -65,146 +93,27 @@ export default class ComboTracker {
...
-      kAbility.Fountain,
-    ]);
-    comboTracker.AddCombo([
-      kAbility.Windmill,
-      kAbility.Bladeshower,
-    ]);
+    const comboTracker = new ComboTracker({
+      player: player,
+      comboBreakers: breakers,
+      comboDelayMs: is5x ? 15000 : 30000,
```
Comment (quisquous) [inline ui/jobs/combo_tracker.ts:114]: Is it possible to not duplicate this time between `base.ts` and `combo_tracker.ts`?

Comment (MaikoTan) [reply ui/jobs/combo_tracker.ts:114]: Yeah, I am just wondering should I put it in constants.ts or just in ComponentManager.

Comment (quisquous) [reply ui/jobs/combo_tracker.ts:114]: I think you could put both values of constants into constants.ts, and then just pick one in each place based on is5x? Maybe there's a better idea, but that's one thought.  I just want to have the numbers in one place.

Comment (MaikoTan) [reply ui/jobs/combo_tracker.ts]: I have the same thought.
But I am thinking "pass is5x to ComboTracker" vs "just pass comboDelayMs that assigned outside to ComboTracker", which is better.

Comment (quisquous) [reply ui/jobs/combo_tracker.ts]: Looks good to me, thanks!

**Pull request merged**